engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetCullFaceMode.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.lwjgl.opengl.GL11;
+import static org.lwjgl.opengl.GL11.GL_BACK;
+import static org.lwjgl.opengl.GL11.GL_FRONT;
+import static org.lwjgl.opengl.GL11.GL_FRONT_AND_BACK;
+import org.terasology.rendering.dag.RenderPipelineTask;
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.dag.tasks.SetCullFaceModeTask;
+
+/**
+ * TODO: Add javadocs
+ */
+public final class SetCullFaceMode implements StateChange {
+
+    private static SetCullFaceMode defaultInstance = new SetCullFaceMode(GL_BACK); // also specified in OpenGL documentation
+    private SetCullFaceModeTask task;
+    private int mode;
+
+    public SetCullFaceMode(int mode) {
+        this.mode = mode;
+        validate();
+    }
+
+    private void validate() {
+        if (mode != GL_BACK
+                && mode != GL_FRONT
+                && mode != GL_FRONT_AND_BACK) {
+            throw new IllegalArgumentException("Mode must be GL_BACK, GL_FRONT or GL_FRONT_AND_BACK.");
+        }
+    }
+
+    public int getMode() {
+        return mode;
+    }
+
+    @Override
+    public StateChange getDefaultInstance() {
+        return defaultInstance;
+    }
+
+    @Override
+    public RenderPipelineTask generateTask() {
+        if (task == null) {
+            task = new SetCullFaceModeTask(mode);
+        }
+
+        return task;
+    }
+
+    @Override
+    public boolean isEqualTo(StateChange stateChange) {
+        if (stateChange instanceof SetCullFaceMode) {
+            return mode == ((SetCullFaceMode) stateChange).getMode();
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isTheDefaultInstance() {
+        return defaultInstance == this;
+    }
+
+    public static String getModeName(int mode) {
```

Comment:
Does this needs to be public? It seems to be used only in the toString() method. If that's the case I would:

1. make this method private
2. remove the input parameter - read `this.mode` instead.
3. remove the switch block and use a static final int,String map instead.

In fact, you could probably remove this whole whole method and in toString write instead:

`return String.format("%21s(%s)", this.getClass().getSimpleName(), modeToStringMap[mode];`